src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -163,10 +164,17 @@ case object ThurloeDatabaseConnector extends DataAccess {
     }
   }
 
-  def set(userKeyValuePair: UserKeyValuePair): Future[DatabaseOperation] = {
-    Aes256Cbc.encrypt(userKeyValuePair.keyValuePair.value.getBytes("UTF-8"), secretKey) match {
-      case Success(encryptedValue) => databaseWrite(userKeyValuePair, encryptedValue)
-      case Failure(x) => Future.failed(x)
+  def set(userKeyValuePairs: UserKeyValuePairs): Future[DatabaseOperation] = {
+    Future.sequence(userKeyValuePairs.toKeyValueSeq.map { userKeyValuePair =>
+      Aes256Cbc.encrypt(userKeyValuePair.keyValuePair.value.getBytes("UTF-8"), secretKey) match {
+        case Success(encryptedValue) => databaseWrite(userKeyValuePair, encryptedValue)
```

Comment:
the problem with this is that each attribute write happens in its own transaction and if one of them fails some of the others may be persisted.

I suggest writing a databaseWrite function the takes key value pairS.